src/uct/tcp/tcp.h

```diff
@@ -98,7 +99,11 @@ typedef enum uct_tcp_ep_conn_state {
      * After it is done, it sends `UCT_TCP_CM_CONN_REQ` to the peer.
      * All AM operations return `UCS_ERR_NO_RESOURCE` error to a caller. */
     UCT_TCP_EP_CONN_STATE_CONNECTING,
-    /* EP is waiting for the magic number acknowledgment. */
+    /* EP is waiting for the magic number acknowledgment. EP is moved to this
+     * state after successful connect(). EP must receive the packet with
+     * `UCT_TCP_CM_CONN_MAGIC_NUMBER_ACK` event before sending `UCT_TCP_CM_CONN_REQ`
+     * in order to make sure that the peer accepted this TCP connection (i.e. it
+     * wasn't dropped by kernel from the accept queue or etc). */
```

Comment:
why needed, won't we will get error in this case anyway?

Reply:
yes, we will get. but it is complex to handle easily. too many cases to handle it - we could receive at different moments.
better to ensure that `connect()` was OK. wdyt?

Reply:
it adds extra round trip, we should aim to complete connection setup as fast as possible..

Reply:
> it adds extra round trip, we should aim to complete connection setup as fast as possible.

yes, will try to avoid it. maybe having `ep_id` in the connection info will be enough - need to check with the failing test